### PR TITLE
BF: make dsi to work with data with multiple b0s etc.

### PR DESCRIPTION
**Pull request by Garyfallidis**

### CODE REVIEW

**Comment by Garyfallidis** [comment]
@arokem this should fix the problem reported by @JohnGriffiths

**Comment by Garyfallidis** [comment]
@JohnGriffiths you may want to try this with the datasets that had multiple b0s and tell us if it works for you. Thx in advance for reporting the problem.

**Comment by Garyfallidis** [comment]
This was also reported at issue #153 from @MrBago.

File: dipy/reconst/tests/test_dsi.py [hunk 1]
```diff
@@ -78,13 +80,18 @@ def test_multib0_dsi():
...
-    assert_equal(new_data.shape[:-1] + (17, 17, 17), PDF.shape)
-    assert_equal(np.alltrue(np.isreal(PDF)), True)
+    ds = DiffusionSpectrumModel(new_gtab)
+    sphere = get_sphere('repulsion724')
+    dsfit = ds.fit(new_data)
+    pdf = dsfit.pdf()
+    odf = dsfit.odf(sphere)
+    assert_equal(new_data.shape[:-1] + (17, 17, 17), pdf.shape)
+    assert_equal(np.alltrue(np.isreal(pdf)), True)
+    from dipy.viz import fvtk
```
Comment (arokem) [inline dipy/reconst/tests/test_dsi.py]: Sorry: I commented on the commit, instead on the diff. 

We should probably remove these viz calls before merging, correct?

**Comment by arokem** [comment]
Actually, let me merge this into my branch, so that @JohnGriffiths can pull it from there, and then I will fix things up there in the test as well.

**Comment by Garyfallidis** [comment]
Yes, please remove the viz parts. Thx for noticing.

**Pull request merged**